sycl/plugins/level_zero/usm_allocator.cpp

```diff
@@ -951,8 +1003,12 @@ USMAllocContext::USMAllocContext(std::unique_ptr<SystemMemory> MemHandle)
     : pImpl(std::make_unique<USMAllocImpl>(std::move(MemHandle))) {}
 
 void *USMAllocContext::allocate(size_t size) {
+  // For full-slab allocations indicates whether slab is from Pool.
   bool FromPool;
-  auto Ptr = pImpl->allocate(size, FromPool);
+  // For chunked allocations indicates whether from existing Slab.
+  // If existing Slab, then FromPool indicates whether slab was from Pool.
+  bool FromSlab;
+  auto Ptr = pImpl->allocate(size, FromSlab, FromPool);
```

Comment:
Why are FromSlab and FromPool being returned here if they are not used after that?

Reply:
The variables just provide storage for the inner function calls. In both "allocate" and "deallocate" they could be moved one level deeper in the call hierarchy.

Reply:
Sure, let's make them local to only where they are needed